### PR TITLE
mesa: Update to 22.0.0

### PR DESCRIPTION
**Pull request by pal1000**
- [swr driver is gone](https://github.com/msys2/MINGW-packages/commit/57f1657ed1202094e0e500926099c6c2f2a1188b)
- [UTF-8 encoding fix for vulkan landed upstream](https://github.com/msys2/MINGW-packages/commit/99720c8c3cd979bfb6653e3b04c23b0927df633c)
- [RADV is available on i686 now](https://github.com/msys2/MINGW-packages/commit/5cf5e308d6fcfc3d86f3b55386ea786e5cf0c834)
- [XXX: We have to explicitly submit LLVM RTTI support status to mesa](https://github.com/msys2/MINGW-packages/commit/108e1574fedc69ad81b28a51ed5f3399446502fb)
- [zstd is required by lavapipe as runtime dependency when available as make dependency](https://github.com/msys2/MINGW-packages/pull/10977/commits/87d7568202138944af11da14c8d208354d7e59e5)
#### Some things that still concern me
- [ ] The updated `0003-win32-Fixes-building-with-msys2-mingw-clang32-clang6.patch` post swr driver removal is a multi-commit e-mail diff patch and I am not sure it applies correctly
- [x] @lygstate has [4 additional patches](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests?scope=all&state=opened&author_username=lygstate) sent upstream for review that may be relevant to us

### CODE REVIEW

**Comment by MehdiChinoune** [comment]
Could you mention that `zstd` is required for Vulkan-lavapipe (It's a runtime dependency) in `optdepends()`

**Comment by pal1000** [comment]
Tried deleting `libzstd.dll` to get lavapipe to fail to load but Vulkan cube demo still worked, so I am not so convinced that zstd is required.

**Comment by lygstate** [comment]
LGTM

File: mingw-w64-mesa/PKGBUILD [hunk 1]
```diff
@@ -12,15 +12,14 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
...
              "${MINGW_PACKAGE_PREFIX}-pkg-config"
-             "${MINGW_PACKAGE_PREFIX}-libelf"
-             "${MINGW_PACKAGE_PREFIX}-zstd")
+             "${MINGW_PACKAGE_PREFIX}-libelf")
 depends=("${MINGW_PACKAGE_PREFIX}-llvm"
          "${MINGW_PACKAGE_PREFIX}-zlib"
          "${MINGW_PACKAGE_PREFIX}-vulkan-loader"
          "${MINGW_PACKAGE_PREFIX}-libsystre"
-         "${MINGW_PACKAGE_PREFIX}-libwinpthread-git")
+         "${MINGW_PACKAGE_PREFIX}-libwinpthread-git"
```
Comment (lygstate) [inline mingw-w64-mesa/PKGBUILD:20]: Seems libwinpthread doesn't need anymore.

Comment (Biswa96) [reply mingw-w64-mesa/PKGBUILD:20]: But if I run `ntldd` command with any DLL file in the package it shows winpthread DLL in dependency.

Comment (lygstate) [reply mingw-w64-mesa/PKGBUILD:20]: gocha, then it's needed

**Pull request merged**